editoast/editoast_derive/src/model/args.rs

```
@@ -57,22 +58,30 @@ impl darling::FromMeta for ErrorArgs {
         let syn::Expr::Path(path) = expr else {
             return Err(darling::Error::custom("expected a type").with_span(expr));
         };
-        Ok(Self::CatchAll(path.path.clone()))
+        Ok(Self::Single(path.path.clone()))
     }
 
     fn from_list(items: &[ast::NestedMeta]) -> darling::Result<Self> {
-        let fine = FineErrorArgs::from_list(items)?;
+        if let Ok(rw) = RwErrorArgs::from_list(items) {
+            return Ok(ErrorArgs::Rw(rw));
+        };
+        let fine = FineErrorArgs::from_list(items).map_err(|err| {
```

I'm not sure that I fully understand the error path in this function. Does the `.from_list()` fails when there is a mix of properties between `read/write` and `create/retrieve/update/delete`? If yes, does it mean that it fails when on line 65 if this is the case, but we ignore the error... and the next try line 68 will again raise an error, and only there, we do something about it? I feel that it would be better express with a three branch match (but at the price of testing always both instead of only one, then fallback to the other). Below is an attempt of expressing what I understood, does this make sense?

```rust
match (RwErrorArgs::from_list(items), FineErrorArgs::from_list(items)) {
  ((Ok(rw), Err(_)) => Ok(ErrorArgs::Rw(rw)),
  ((Err(_), Ok(fine)) => Ok(ErrorArgs::Fine(fine)),
  ((Ok(_), Ok(_)) => unreachable!("both branch cannot be parsed successfully at the same time since properties of each are disjoint sets"),
  ((Err(e), Err(_)) => Err(e.note("bblablabla")),
}
```

---

I did it that way to avoid parsing the meta list twice on non-error paths. But I much prefer this style 0276306aa